src/Symfony/Component/Routing/Route.php

```diff
@@ -567,7 +582,7 @@ public function compile()
 
     private function sanitizeRequirement($key, $regex)
     {
-        if (!is_string($regex)) {
+        if (!is_string($regex) && $this->enforceRequirementAsString) {
```

Comment:
Is it still a regex if it's not a string? If not, the local parameter should be renamed imo